samza-api/src/main/java/org/apache/samza/task/StreamOperatorTaskFactory.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.task;
+
+import org.apache.samza.annotation.InterfaceStability;
+
+
+/**
+ * Build {@link AsyncStreamTask} instances.
+ * <p>
+ * Implementations should return a new instance of {@link AsyncStreamTask} for each {@link #createInstance()} invocation.
+ */
+@InterfaceStability.Evolving
+public interface StreamOperatorTaskFactory extends TaskFactory<AsyncStreamTask> {
```

Comment:
Does this need to be in samza-api? Seems like it should be possible to do this in core, and I don't think we need to expose this to users.

Reply:
Agreed. This is will go away with the InternalTaskContext refactor.
Moved it to samza-core.